pkg/resources/manager/manager.go

```diff
@@ -202,24 +164,47 @@ func (r *resourceManager) initialize(ctx context.Context, machines []*api.Machin
 
 	r.ctx = ctx
 
-	err := r.loadTotalResources()
-	if err != nil {
-		return err
+	// Initialize all sources and check for common resources
+	var initializedResources *set.Set
+	for _, s := range r.sources {
+		resources, err := s.Init(r.ctx)
+		if err != nil {
+			return err
+		}
+
+		if initializedResources == nil {
+			initializedResources = resources
+		} else {
```

Comment:
this else block isn't needed

Reply:
It is needed, else in the first iteration, it will have all the common resources